dev-tools/xtask/src/virtual_hardware.rs

```diff
@@ -270,7 +270,13 @@ fn remove_softnpu_zone(npu_zone: &Utf8Path) -> Result<()> {
         "--ports",
         "sc0_1,tfportqsfp0_0",
     ]);
-    execute(cmd)?;
+    if let Err(output) = execute(cmd) {
+        // Don't throw an error if the zone was already removed
+        if output.to_string().contains("No such zone configured") {
```

Comment:
It seems like we should be returning an error if the conditional is false, right? This will fall through to `return Ok(())`, unless I'm missing something.

Reply:
I was under the impression this was by design. We want to keep destroying things even if there is an error destroying the zone. Or at least that was my take on it

Reply:
But I guess perhaps we do want to print out the error message so people know something didn't quite go the way we wanted it to.

Reply:
I think it depends on the error. If we failed to remove it because it's already gone, then yes, we should continue. If we failed to remove it for an unexpected reason, then I'm not so sure. I'd at least want to know _why_ it failed, even if we ultimately decide to continue.

Reply:
whoops, silly oversight -- agreed, I need to actually propagate most errors. updated!